services/galley/src/Galley/API/Query.hs

```diff
@@ -171,31 +168,16 @@ getGlobalTeamConversation ::
   Sem r Public.GlobalTeamConversation
 getGlobalTeamConversation lusr cid tid = do
   let uid = tUnqualified lusr
+      ltid = qualifyAs lusr tid
   void $ noteS @'NotATeamMember =<< E.getTeamMember tid (tUnqualified lusr)
-  E.getGlobalTeamConversation tid >>= \case
+  E.getGlobalTeamConversation ltid >>= \case
     Nothing -> do
-      conv <- E.createGlobalTeamConversation (qualifyAs lusr tid) uid
-      mlsData <- case Conv.convProtocol conv of
-        ProtocolMLS mls -> pure mls
-        ProtocolProteus -> throw (InternalErrorWithDescription "Wrong protocol, expected MLS, got Proteus.")
-      -- FUTUREWORK: remove this. we are planning to remove the need for a nullKeyPackageRef
-      let convId = Conv.convId conv
-          lconv = qualifyAs lusr convId
-      E.addMLSClients lconv (qUntagged lusr) (Set.singleton (cid, nullKeyPackageRef))
-      pure $
-        GlobalTeamConversation
-          (qUntagged lconv)
-          (Conv.convMetadata conv)
-          mlsData
-    Just conv -> do
-      mlsData <- case Conv.convProtocol conv of
-        ProtocolMLS mls -> pure mls
-        ProtocolProteus -> throw (InternalErrorWithDescription "Wrong protocol, expected MLS, got Proteus.")
-      pure $
-        GlobalTeamConversation
-          (qUntagged . qualifyAs lusr $ Conv.convId conv)
-          (Conv.convMetadata conv)
-          mlsData
+      gtc <- E.createGlobalTeamConversation ltid uid
+      E.addMLSClients (localGtcId gtc) (qUntagged lusr) (Set.fromList [(cid, nullKeyPackageRef)])
```

Comment:
You're still setting the client on on GET, which is prone to race conditions. The `nullKeyPackageRef` hack should not be used anymore.

Reply:
I think this was a rebase mix-up. Good catch.

Reply:
Oh, more like I missed a spot.